### ISSUE TITLE
cupy.linalg.svd fail to compute svd for large matrix

### ISSUE THREAD

**Issue by LUOXIAO92** [cupy/cupy#7611]
### Description

cupy.linalg.svd fail to compute svd when the size of matrix larger than $18900 \times 18900$. 
As the following code, I make a large size of hermitian matrix $A$ and compute svd. Then print the singular values and error $\frac{||A-U\Sigma V^\dagger||}{||A||}$. In the case of $18900 \times 18900$ matrix, the svd can be computed successfully and the singular values are all positive, and the error is about $10^{-14}$. The computation time with nvidia H100 is about 304s. 
However, when the matrix is the size of $19000 \times 19000$, the computation fails, and the computation time is about 20~60s, which is more shorter than the successful case. And the singular values have negative values, the error also becomes about 0.707 or $1/\sqrt(2)$

### To Reproduce

```py
import time
import cupy as cp

if __name__ == "__main__":

    N = 19_000
    rs = cp.random.RandomState(seed=1234)
    A = rs.standard_normal(size=(N,N))+1j*rs.standard_normal(size=(N,N))
    A = A@cp.conj(A.T)
    A = A / cp.max(cp.abs(A))
    
    print("A",A.shape, A.nbytes/2**30,"Gbytes")

    t0 = time.time()
    u, s, vh = cp.linalg.svd(A)
    t1 = time.time()
    

    print("svd time= {:.2f}".format(t1-t0))
    print("s",s)

    A2 = cp.einsum("ij,j,jk->ik",u,s,vh)

    err = cp.linalg.norm(A-A2)/cp.linalg.norm(A)
    print("err(A-A2)",err)
    print()
    del A, u, s, vh
```

### Installation

Conda-Forge (`conda install ...`)

### Environment

```
OS                           : Linux-5.4.0-144-generic-x86_64-with-glibc2.31
Python Version               : 3.10.10
CuPy Version                 : 12.0.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.24.2
SciPy Version                : 1.10.1
Cython Build Version         : 0.29.33
Cython Runtime Version       : None
CUDA Root                    : /home/.conda/envs/py3.10
nvcc PATH                    : None
CUDA Build Version           : 11020
CUDA Driver Version          : 12000
CUDA Runtime Version         : 11080
cuBLAS Version               : (available)
cuFFT Version                : 10900
cuRAND Version               : 10300
cuSOLVER Version             : (11, 4, 1)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 8)
Thrust Version               : 101000
CUB Build Version            : 101000
Jitify Build Version         : ed53a41
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : 21403
NCCL Runtime Version         : 21701
cuTENSOR Version             : 10700
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA H100 PCIe
Device 0 Compute Capability  : 90
Device 0 PCI Bus ID          : 0000:16:00.0
```

### Additional Information

conda environment :
```
# packages in environment at /home/.conda/envs/py3.10:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py310h5764c6d_1005    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2022.12.7            ha878542_0    conda-forge
certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h255011f_3    conda-forge
charset-normalizer        3.1.0              pyhd8ed1ab_0    conda-forge
comm                      0.1.3              pyhd8ed1ab_0    conda-forge
cryptography              40.0.1          py310h34c0648_0    conda-forge
cudatoolkit               11.8.0              h37601d7_11    conda-forge
cupy                      12.0.0          py310h9216885_0    conda-forge
cuquantum                 23.03.0.20           h0bbbad9_0    conda-forge
cuquantum-python          23.03.0         py310hbe3ad0c_0    conda-forge
custatevec                1.3.0                h0800d71_0    conda-forge
cutensor                  1.7.0.1              h12f7317_0    conda-forge
cutensornet               2.1.0           nompi_h0800d71_100    conda-forge
debugpy                   1.6.7           py310heca2aa9_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
fastrlock                 0.8             py310hd8f1fbe_3    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
importlib-metadata        6.4.1              pyha770c72_0    conda-forge
importlib_metadata        6.4.1                hd8ed1ab_0    conda-forge
ipykernel                 6.22.0             pyh210e3f2_0    conda-forge
ipython                   8.12.0             pyh41d4057_0    conda-forge
jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
jupyter_client            8.2.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.3.0           py310hff52083_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libblas                   3.9.0           16_linux64_openblas    conda-forge
libcblas                  3.9.0           16_linux64_openblas    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 12.2.0              h65d4601_19    conda-forge
libgfortran-ng            12.2.0              h69a702a_19    conda-forge
libgfortran5              12.2.0              h337968e_19    conda-forge
libgomp                   12.2.0              h65d4601_19    conda-forge
liblapack                 3.9.0           16_linux64_openblas    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.21          pthreads_h78a6416_3    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libsqlite                 3.40.0               h753d276_0    conda-forge
libstdcxx-ng              12.2.0              h46fd767_19    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libzlib                   1.2.13               h166bdaf_4    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mpi4py                    3.1.4                    pypi_0    pypi
ncurses                   6.3                  h27087fc_1    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
numpy                     1.24.2          py310h8deb116_0    conda-forge
openssl                   3.1.0                h0b41bf4_0    conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 23.0               pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pip                       23.0.1             pyhd8ed1ab_0    conda-forge
platformdirs              3.2.0              pyhd8ed1ab_0    conda-forge
pooch                     1.7.0              pyha770c72_3    conda-forge
prompt-toolkit            3.0.38             pyha770c72_0    conda-forge
prompt_toolkit            3.0.38               hd8ed1ab_0    conda-forge
psutil                    5.9.4           py310h5764c6d_0    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygments                  2.15.0             pyhd8ed1ab_0    conda-forge
pyopenssl                 23.1.1             pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.10         he550d4f_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    3_cp310    conda-forge
pyzmq                     25.0.2          py310h059b190_0    conda-forge
readline                  8.2                  h8228510_1    conda-forge
requests                  2.28.2             pyhd8ed1ab_1    conda-forge
scipy                     1.10.1          py310h8deb116_0    conda-forge
setuptools                67.6.1             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
tornado                   6.2             py310h5764c6d_1    conda-forge
traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
typing-extensions         4.5.0                hd8ed1ab_0    conda-forge
typing_extensions         4.5.0              pyha770c72_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   1.26.15            pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
wheel                     0.40.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zipp                      3.15.0             pyhd8ed1ab_0    conda-forge
```

**Comment by leofang** [comment]
> However, when the matrix is the size of 19000 x 19000, the computation fails,

Could you share the full output from your snippet? In what way it failed?

**Comment by LUOXIAO92** [comment]
> > However, when the matrix is the size of 19000 x 19000, the computation fails,
> 
> Could you share the full output from your snippet? In what way it failed?

In the case of $18900 \times 18900$ matrix
```
A (18900, 18900) 5.322843790054321 Gbytes
svd time= 304.78
s [3.89840399e+00 3.88786835e+00 3.88024928e+00 ... 3.61865031e-08
 2.01511721e-08 2.77994076e-10]
err(A-A2) 4.0673748862790336e-14
```

In the case of $19000 \times 19000$ matrix:
```
#hermitian matrix
A (19000, 19000) 5.379319190979004 Gbytes
svd time= 46.39
s [-1.37425444e+00  1.78301021e+00  1.83194712e+00 ... -6.90348034e-08
  4.30540599e-08 -1.87181715e-09]
err(A-A2) 0.7072679940244606

#non-hermitian matrix
A (19000, 19000) 5.379319190979004 Gbytes
svd time= 60.25
s [-30.70363106 -30.50898721 -30.76398079 ...   0.25509631   0.28270966
  -0.22002994]
err(A-A2) 0.7070857775516758

#non-symmetric matrix
A (19000, 19000) 2.689659595489502 Gbytes
svd time= 24.87
s [-21.58806333  21.73383556  21.54055944 ...  -0.45973311   0.14808218
  -0.07805201]
err(A-A2) 0.7070618687145447

#symmetric matrix
A (19000, 19000) 2.689659595489502 Gbytes
svd time= 23.00
s [-1.34770060e+00  1.75792709e+00  1.85483171e+00 ... -6.39524100e-08
  1.37397069e-08 -2.69810386e-08]
err(A-A2) 0.7072586993273443
```

**Comment by leofang** [comment]
Sorry I dropped the ball, @LUOXIAO92. I can reproduce the issue on an H100. 

The cuSOLVER log looked fine to me (`CUSOLVERDN_LOG_LEVEL=5 python my_script.py`), so I checked the returned info array, and there was an error that's ignored by default by CuPy. It can be seen if you instead import `cupyx` and do
```python
with cupyx.errstate(linalg='raise'):  # this mimics the usage of numpy.errstate()
    t0 = time.time()
    u, s, vh = cp.linalg.svd(A)
    t1 = time.time()
```
and you'd get
```
A (19000, 19000) 5.379319190979004 Gbytes

(...)

numpy.linalg.LinAlgError: Error reported by zgesvd in cuSOLVER. devInfo = [18999]. Please refer to the cuSOLVER documentation.
```
Will reach out to the cuSOLVER team tmr.

**Comment by LUOXIAO92** [comment]
Thank you. So should I contact the cuSOLVER team?

**Comment by leofang** [comment]
No worries, I will.

**Comment by leofang** [comment]
Thanks for reporting @LUOXIAO92. The team will investigate it. Tracked in nvbugs 4175628.

**Comment by leofang** [comment]
> Thanks for reporting @LUOXIAO92. The team will investigate it. Tracked in nvbugs 4175628.

Update: This has been fixed by the cuSOLVER team in CUDA 12.2.2:
https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cusolver-release-12-2-update-2

@LUOXIAO92 please feel free to give it a try and let us know if you still see any issues. Thanks again for reporting the problem to CuPy and NVIDIA.
